packages/grid_client/src/clients/tf-grid/contracts.ts

```diff
@@ -325,6 +335,25 @@ class TFContracts extends Contracts {
     return ids;
   }
 
+  async batchUnlockContracts(ids: number[]) {
+    const extrinsics: ExtrinsicResult<Contract>[] = [];
+    for (const id of ids) {
+      extrinsics.push(await this.unlock(id));
+    }
+    await this.client.applyAllExtrinsics(extrinsics);
+    return ids;
+  }
+  async unlockMyContracts(graphqlURL: string) {
+    const contracts = await this.listMyContracts({
+      stateList: [ContractStates.GracePeriod],
+      graphqlURL,
+    });
+    const ids: number[] = [];
+    for (const contract of contracts.nameContracts) ids.push(parseInt(contract.contractID));
+    for (const contract of contracts.nodeContracts) ids.push(parseInt(contract.contractID));
+    for (const contract of contracts.rentContracts) ids.push(parseInt(contract.contractID));
```

Comment:
@0oM4R @Mahmoud-Emad 
We could also do a nested loop to avoid multi loops and copying data again

```ts
// Example

const ids: number[] = [];
for (const contractType in contracts) {
  for (const { contractID } of contracts[contractType]) {
    ids.push(contractID)
  }
}
```

Reply:
@MohamedElmdary @Mahmoud-Emad please check the updated code, and tell me if it need to be enhanced 
https://github.com/threefoldtech/tfgrid-sdk-ts/blob/27421b70f6ffa394919c96f4d217aa0a681c0f9f/packages/grid_client/src/clients/tf-grid/contracts.ts#L345-L352